equinix/data_source_metal_device.go

```diff
@@ -232,26 +237,28 @@ func dataSourceMetalDeviceRead(d *schema.ResourceData, meta interface{}) error {
 	} else {
 		deviceId := deviceIdRaw.(string)
 		var err error
-		device, _, err = client.Devices.Get(deviceId, deviceReadOptions)
+		device, _, err = client.DevicesApi.FindDeviceById(nil, deviceId).Execute() // deviceReadOptions ?
 		if err != nil {
 			return err
 		}
 	}
 
 	d.Set("hostname", device.Hostname)
-	d.Set("project_id", device.Project.ID)
-	d.Set("device_id", device.ID)
+	d.Set("project_id", proj.Id)
+	d.Set("device_id", device.Id)
 	d.Set("plan", device.Plan.Slug)
 	d.Set("facility", device.Facility.Code)
 	if device.Metro != nil {
-		d.Set("metro", strings.ToLower(device.Metro.Code))
+		d.Set("metro", strings.ToLower(*device.Metro.Code))
 	}
-	d.Set("operating_system", device.OS.Slug)
+	d.Set("operating_system", device.OperatingSystem.Slug)
 	d.Set("state", device.State)
 	d.Set("billing_cycle", device.BillingCycle)
-	d.Set("ipxe_script_url", device.IPXEScriptURL)
-	d.Set("always_pxe", device.AlwaysPXE)
+	d.Set("ipxe_script_url", device.IpxeScriptUrl)
```

Comment:
Little changes like these in properties between Packngo and MetalGo make me wonder if we have the 'better' pattern in one project or the other. If "ID" "PXE" "URL" are preferable, the codegen in metal-go could consider take that route with initialisms (https://github.com/OpenAPITools/openapi-generator/issues/4833 has an associated stale/open PR)

Reply:
If/when the generator has broad support for defining acronyms/initialisms, it may be worth looking into proper capitalization of those.  At the moment, the camel-casing is consistent across `metal-go`, `metal-java`, and `metal-python`, so I'd be hesitant to add a bunch of post-patches in one of those (or, even worse, maintain equivalent post-patches for all of them) to change that.

Casing decisions for model names are driven by the spec, and I think that's where we could use more consistency; here's an example where BGP is capitalized differently because of the spec:

- https://github.com/equinix-labs/metal-go/blob/7e28dc6f080125526593528336c15de56e0e99bf/metal/v1/model_bgp_session.go
- https://github.com/equinix-labs/metal-go/blob/7e28dc6f080125526593528336c15de56e0e99bf/metal/v1/model_bgp_session_input.go

At this point, I'd probably lean towards camel-casing acronyms & initialisms in model names in the spec since that's what the generator would do for property names that contain the same acronyms & initialisms.

Reply:
Looks like initialism may be fixed. 
https://github.com/OpenAPITools/openapi-generator/pull/15083

Reply:
I added equinix-labs/metal-go#112 to investigate the `reservedWordsMapping` and see if it supports initialisms without introducing unintended side effects.  Given that casing for initialisms is a stylistic issue rather than a functional one, I think it would be ok to adopt metal-go in this project without waiting for the results of that SDK investigation.